### PR TITLE
BTE/GPS android product flavor setup

### PR DESCRIPTION
**Pull request by mattThousand**
This commit:
- Introduces workflows for building and deploying the BTE version of the android
app equivalent to the the existing workflows for the GPS app.

- Sets the triggering branch for publishing to be `bte-release`.

- Updates the `Fastfile` for distributing android `gps` and `bte` product flavors. 

- Adds a native module to determine `tracingStrategy` based on product flavor (`bte` or `gps`)

### CODE REVIEW

File: android/fastlane/Fastfile [hunk 1]
```diff
@@ -23,59 +23,98 @@ platform :android do
...
 
-  desc "Build a Release APK and AAB"
-  lane :release do
+  desc "GPS: Build a Release AAB"
+  lane :release_gps do
 
-    increment_version_code(
-      gradle_file_path: "../android/app/build.gradle",
-    )
+    build_number =  ENV["GITHUB_RUN_NUMBER"]
```
Comment (mattThousand) [inline android/fastlane/Fastfile:37]: The idea here is to set the version code automatically from the workflow run number. `VersionName` updates are infrequent and can be done manually in `build.gradle`

File: android/app/src/main/java/org/pathcheck/covidsafepaths/util/COVIDSafePathsConfig.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package org.pathcheck.covidsafepaths.util
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import org.pathcheck.covidsafepaths.BuildConfig
+
+class COVIDSafePathsConfig(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {
+
+    override fun getName(): String {
```
Comment (johnschoeman) [inline android/app/src/main/java/org/pathcheck/covidsafepaths/util/COVIDSafePathsConfig.kt:9]: minor thing, but not sure we are ever going to need this function?

File: android/fastlane/Fastfile (hunk 1 above)
Comment (johnschoeman) [inline android/fastlane/Fastfile:37]: I dig this! i would double check that its not conflicting with some of the work that is being done here: https://github.com/Path-Check/covid-safe-paths/pull/873

let me know if you wanna chat through it some more, but Id love to get the buildnumbers for the ios side incrementing and independent of the version number as well, its a bit odd to have the version number change along with the build number and means that we have many many versions.

File: .github/workflows/staging-android-bte.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+name: Staging Android
```
Comment (johnschoeman) [inline .github/workflows/staging-android-bte.yml]:
```suggestion
name: Staging Android BTE
```

File: .github/workflows/staging-android-bte.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+      - release-candidate
```
Comment (johnschoeman) [inline .github/workflows/staging-android-bte.yml]:
```suggestion

```

**Review by johnschoeman — DISMISSED** [review]
Looking good to me!  A few minor minor comments, if youre interested.

**Comment by johnschoeman** [comment]
> @mattThousand mattThousand dismissed johnschoeman’s stale review via 7a37b05 34 seconds ago

oh man i hate that this happens on every code push, ping me when its ready to go and i can re-review!

File: android/app/src/main/java/org/pathcheck/covidsafepaths/util/COVIDSafePathsConfig.kt (hunk 2 above)
Comment (mattThousand) [reply android/app/src/main/java/org/pathcheck/covidsafepaths/util/COVIDSafePathsConfig.kt:9]: good question: it's used to expose the name of the module to the `JS` layer, so we are stuck with it I believe

File: android/fastlane/Fastfile (hunk 1 above)
Comment (mattThousand) [reply android/fastlane/Fastfile:37]: I like @tstirrat's approach in #873 - we'd read the new version name from `package.json`. The `versionCode` on android and the `build number` on iOS should be independent, and increment automatically on each workflow run. What do you think?

Comment (johnschoeman) [reply android/fastlane/Fastfile:37]: tbh, i think we could be manually upping the build numbers, that said, having some automation is nice and if tim's strategy works then it works!

**Review by johnschoeman — APPROVED** [review]
💯 🚢

File: android/fastlane/Fastfile (hunk 1 above)
Comment (mattThousand) [reply android/fastlane/Fastfile:37]: sounds good! this PR won't address versionName; in the interest of time we can keep the versionCode incrementation logic here and we can all discuss the versionName piece or revisit this question as part of #873

Comment (tstirrat) [reply android/fastlane/Fastfile:37]: Yeah, version codes can be whaterver as long as they always increase. Seems good.

File: android/fastlane/Fastfile [hunk 5]
```diff
@@ -23,59 +23,96 @@ platform :android do
...
+  end 
+
+  desc "Build a Release AAB"
+  lane :release_bte do
+
+    build_number =  ENV["GITHUB_RUN_NUMBER"]
+
+    increment_version_code(
+      gradle_file_path: "app/build.gradle",
+      version_code: build_number
```
Comment (tstirrat) [inline android/fastlane/Fastfile:85]: This would have been clearer as `github_run_number`

**Pull request merged**